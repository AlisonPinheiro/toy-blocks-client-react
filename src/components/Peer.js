import React from 'react';
import {peer} from '../types';
import styled from 'styled-components';
import colors from '../constants/colors';
import Status from './Status';

const Wrapper = styled.div`
  margin: 10px 0;
  padding: 10px;
  border: 1px solid #aaaaaa;
  background: ${colors.contentBackground};
  border-radius: 5px;
`;

const Head = styled.div`
  display: flex;
  justify-content: space-between;
`;

const Body = styled.div`
  display: flex;
  justify-content: space-between;
`;

const Name = styled.div`
  margin-bottom: 6px;
  color: ${colors.primary};
  font-size: 16px;
  font-weight: 700;
`;

const URL = styled.div`
  margin-top: 2px;
  margin-bottom: 2px;
  color: ${colors.faded};
  font-weight: 500;
`;

const Peer = ({peer}) => (
  <Wrapper>
    <Head>
      <Name>
        {peer.name || 'Unknown'}
      </Name>
    </Head>
    <Body>
      <URL>
        {peer.url}
      </URL>
      <Status {...peer} />
    </Body>
  </Wrapper>
);

Peer.propTypes = {
  peer: peer.isRequired
};

export default Peer;
