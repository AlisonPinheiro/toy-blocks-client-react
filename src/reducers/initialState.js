export default {
  peers: {
    list: [
      {
        url: 'http://localhost:3002',
        online: false,
        name: 'Node 1',
        loading: false
      },
      {
        url: 'http://localhost:3003',
        online: false,
        name: 'Node 2',
        loading: false
      },
      {
        url: 'http://localhost:3004',
        online: false,
        name: 'Node 3',
        loading: false
      },
      {
        url: 'http://localhost:3005',
        online: false,
        name: 'Node 4',
        loading: false
      }
    ]
  }
};
