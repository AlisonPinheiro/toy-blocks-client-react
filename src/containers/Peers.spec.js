import React from "react";
import { shallow } from "enzyme";
import configureMockStore from "redux-mock-store";
import thunk from 'redux-thunk';
import { Provider } from "react-redux";
import { create } from "react-test-renderer";
import ConnectedPeers, { Peers } from "./Peers";
import Peer from "../components/Peer";
import initialState from "../reducers/initialState";

describe("<Peers />", () => {
  const actions = {
    checkPeerStatuses: jest.fn()
  };

  it("should contain <Peer />", () => {
    const wrapper = shallow(
      <Peers
        actions={actions}
        peers={initialState.peers}
      />
    );

    expect(wrapper.find(Peer).length).toEqual(4);
  });

  // it("calls saveFuelSavings upon clicking save", () => {
  //   const wrapper = mount(
  //     <FuelSavingsPage
  //       actions={actions}
  //       fuelSavings={initialState.fuelSavings}
  //     />
  //   );
  //
  //   const save = wrapper.find('input[type="submit"]');
  //   save.simulate("click");
  //
  //   expect(actions.saveFuelSavings).toHaveBeenCalledWith(
  //     initialState.fuelSavings
  //   );
  // });
  //
  // it("calls calculateFuelSavings upon changing a field", () => {
  //   const wrapper = mount(
  //     <FuelSavingsPage
  //       actions={actions}
  //       fuelSavings={initialState.fuelSavings}
  //     />
  //   );
  //   const name = "newMpg";
  //   const value = 10;
  //
  //   const input = wrapper.find('input[name="newMpg"]');
  //   input.simulate("change", { target: { name, value } });
  //
  //   expect(actions.calculateFuelSavings).toHaveBeenCalledWith(
  //     initialState.fuelSavings,
  //     name,
  //     value
  //   );
  // });

  it("should match snapshot", () => {
    const middlewares = [thunk];
    const store = configureMockStore(middlewares)(initialState);
    const component = create(
      <Provider store={store}>
        <ConnectedPeers />
      </Provider>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
