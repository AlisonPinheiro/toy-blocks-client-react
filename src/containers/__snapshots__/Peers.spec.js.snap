// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Peers /> should match snapshot 1`] = `
<div>
  <h1>
    Peers
  </h1>
  <div
    className="sc-htpNat cAglkR"
  >
    <div
      className="sc-bxivhb gSSgMV"
    >
      <div
        className="sc-EHOje fvoECt"
      >
        Node 1
      </div>
    </div>
    <div
      className="sc-ifAKCX fOGIFb"
    >
      <div
        className="sc-bZQynM dSZNti"
      >
        http://localhost:3002
      </div>
      <div
        className="sc-bdVaJa ffSlWY"
        color="danger"
      >
        Offline
      </div>
    </div>
  </div>
  <div
    className="sc-htpNat cAglkR"
  >
    <div
      className="sc-bxivhb gSSgMV"
    >
      <div
        className="sc-EHOje fvoECt"
      >
        Node 2
      </div>
    </div>
    <div
      className="sc-ifAKCX fOGIFb"
    >
      <div
        className="sc-bZQynM dSZNti"
      >
        http://localhost:3003
      </div>
      <div
        className="sc-bdVaJa ffSlWY"
        color="danger"
      >
        Offline
      </div>
    </div>
  </div>
  <div
    className="sc-htpNat cAglkR"
  >
    <div
      className="sc-bxivhb gSSgMV"
    >
      <div
        className="sc-EHOje fvoECt"
      >
        Node 3
      </div>
    </div>
    <div
      className="sc-ifAKCX fOGIFb"
    >
      <div
        className="sc-bZQynM dSZNti"
      >
        http://localhost:3004
      </div>
      <div
        className="sc-bdVaJa ffSlWY"
        color="danger"
      >
        Offline
      </div>
    </div>
  </div>
  <div
    className="sc-htpNat cAglkR"
  >
    <div
      className="sc-bxivhb gSSgMV"
    >
      <div
        className="sc-EHOje fvoECt"
      >
        Node 4
      </div>
    </div>
    <div
      className="sc-ifAKCX fOGIFb"
    >
      <div
        className="sc-bZQynM dSZNti"
      >
        http://localhost:3005
      </div>
      <div
        className="sc-bdVaJa ffSlWY"
        color="danger"
      >
        Offline
      </div>
    </div>
  </div>
</div>
`;
